enumeratum-json4s/src/main/scala/enumeratum/values/Json4s.scala

```diff
@@ -0,0 +1,67 @@
+package enumeratum.values
+
+import org.json4s.CustomSerializer
+import org.json4s.JsonAST.{JInt, JLong, JString}
+
+
+object Json4s {
+
+  def serializer[A <: IntEnumEntry: Manifest](enum: IntEnum[A]): CustomSerializer[A] = new CustomSerializer[A](_ => (
+    {
+      case JInt(i) if enum.withValueOpt(i.toInt).isDefined => enum.withValue(i.toInt)
+      case JLong(i) if enum.withValueOpt(i.toInt).isDefined => enum.withValue(i.toInt)
+    },
+    {
+      case x: A => JLong(x.value)
+    }
+  ))
+
+  def serializer[A <: LongEnumEntry: Manifest](enum: LongEnum[A]): CustomSerializer[A] = new CustomSerializer[A](_ => (
+    {
+      case JInt(i) if enum.withValueOpt(i.toLong).isDefined => enum.withValue(i.toLong)
+      case JLong(i) if enum.withValueOpt(i).isDefined => enum.withValue(i)
+    },
+    {
+      case x: A => JLong(x.value)
+    }
+  ))
+
+  def serializer[A <: ShortEnumEntry: Manifest](enum: ShortEnum[A]): CustomSerializer[A] = new CustomSerializer[A](_ => (
+    {
+      case JInt(i) if enum.withValueOpt(i.toShort).isDefined => enum.withValue(i.toShort)
```

Comment:
Here too, we should have guards in case the numbers are too big to fit inside the Short range